pkg/porter/list.go

```diff
@@ -102,6 +112,13 @@ type DisplayInstallation struct {
 	// Status of the installation.
 	Status                      storage.InstallationStatus `json:"status,omitempty" yaml:"status,omitempty" toml:"status,omitempty"`
 	DisplayInstallationMetadata `json:"_calculated" yaml:"_calculated"`
+
+	// DisplayInstallationState is the latest state of the installation.
+	// It is either "installed", "uninstalled", or "defined".
+	DisplayInstallationState string `json:"displayInstallationState,omitempty" yaml:"displayInstallationState,omitempty" toml:"displayInstallationState,omitempty"`
```

Comment:
Can you move these two new fields to the DisplayInstallationMetadata data structure? That is where all calculated values for the display of the installation are being put, so that users can tell what is part of the installation's definition, vs info that porter calculated for its display in that command.